sdk/openai/openai/src/rest/models.ts

```diff
@@ -970,6 +970,24 @@ export interface EmbeddingsOptions {
   dimensions?: number;
 }
 
+/** A response containing error details. */
+export interface ErrorResponse {
+  /** The error object. */
+  error: OpenAIErrorModel;
+}
```

Comment:
This is the model used in the rest layer for the service error response. This is used instead of the `ErrorResponse` model from Azure core so that we can inject the error property to be OpenAI errors.

Reply:
The problem is that this model is not accurate because it expects an `error` field in the body which is not true for OpenAI error bodies. I think you need to make this model an alias to `OpenAIErrorModel` instead.

Reply:
That's interesting. I was referring to our `getEmbeddings` recording with a captured OpenAI error [here](https://github.com/Azure/azure-sdk-assets/blob/js/openai/openai_3ad05036f3/js/sdk/openai/openai/recordings/node/openai_openaikey_client_getembeddings/recording_wrong_prompt_type.json#L40C2-L45C23) and there's an `error` field in the body. Can you provide more context on OpenAI error bodies?

Reply:
Ah I see, my bad. In this case, the two error shapes are more compatible than I initially thought. Then what you have here makes sense to me.